### ISSUE TITLE
Weird Tab behavior

### ISSUE THREAD

**Issue by blitz-research** [engor/Ted2Go#48]
Enter the following in an empty file with auto indent disabled (haven't tested with enabled)...

```
Method Test
Local test
```

After you've typed the last 'test' go to the start of line and hit 'tab'. Here, I end up with this:

```
Method TTestArgsLocal test
```

Very strange!

**Comment by engor** [comment]
Latest dev branch or master? I can't get that result.

**Comment by blitz-research** [comment]
Ok, it's a master only issue.

I'll try and use dev branch from now until release time.

**Comment by blitz-research** [comment]
Can reproduce this in dev now.

Here, when I enter:

```
Class Test
Local te
```

The autocomplete thing pops up (not sure why).

If you hit 'home' to go to start of line with autocomplete still open, it stays open, and if you then hit tab, you end up with:

```
Class Testime te
```

The problem seems to be caused by going to the start of line with autocomplete open.

**Comment by engor** [comment]
Now, completion list will hide by pressing Home/End keys.
Pushed to dev and master.

> The autocomplete thing pops up (not sure why).

Completion list based on 'using' directive to filter unnecessary results.
If we have locked file - then completion will look through all files imported to locked for searching all usings. Plus usings in current file.
If have no locked file - check usings only in current file.

an empty file w/o locked any file
![completion1](https://user-images.githubusercontent.com/5233803/30679470-02dd5d12-9ec4-11e7-8a80-50033c30f214.png)

an empty file with locked Ted2.monkey2
![completion2](https://user-images.githubusercontent.com/5233803/30679477-235cecc4-9ec4-11e7-9f99-7ffa0eeaacfe.png)
